crates/aptos/src/common/types.rs

```diff
@@ -294,9 +304,9 @@ pub struct ProfileSummary {
     pub network: Option<Network>,
     pub has_private_key: bool,
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub public_key: Option<Ed25519PublicKey>,
+    pub public_key: Option<String>,
```

Comment:
Sad that we're losing type safety here. Is there a good reason?

Reply:
This is for a string output?

I'm not sure there's a reason to worry about type safety here?

Reply:
Usually good to maintain type safety: readability, consistency (e.g., changing a trait on Ed25519PublicKey will effect the change everywhere)

Reply:
Yeah, normally I would agree, but in this case I have two options:
1. Add a lot of these extra annotations (as you see in the profile config)
2. Convert this to a string here.

Key here is that it is for display purposes, I chose this here, but I guess it can be consistent

Reply:
Okay, I remember now.

The serialization format (in JSON) for addresses is particularly annoying, it has no 0x in the front, and is inconsistent with other places.  Unfortunately, I don't want to change serialization formats, so I changed it to a string.

For Public key, similar thing, though that can be done with the similar functions for the profile.